mohawk/base.py

```diff
@@ -60,20 +60,25 @@ def _authorize(self, mac_type, parsed_header, resource,
                               'theirs: {theirs}'
                               .format(ours=mac, theirs=their_mac))
 
-        if 'hash' not in parsed_header and accept_untrusted_content:
+        check_hash = True
+        if 'hash' not in parsed_header:
             # The request did not hash its content.
-            log.debug('NOT calculating/verifiying payload hash '
-                      '(no hash in header)')
-            check_hash = False
-            content_hash = None
-        else:
-            check_hash = True
-            content_hash = resource.gen_content_hash()
+            if accept_untrusted_content:
+                log.debug('NOT calculating/verifiying payload hash '
+                          '(no hash in header)')
+                check_hash = False
+                content_hash = None
+            elif (resource.content is EmptyValue):
```

Comment:
`EmptyValue` means that `content` has not been given an explicit value. What you meant was to look for actual empty values, such as an empty string or `None`.

We need to protect against someone defining a receiver object and *forgetting* to pass in the `content` keyword. 

For example, the following code contains a terrible typo that would allow content tampering, something that the developer did not intend.

```python
request = {
  'headers': {
    'Authorization': sender.request_header,
    'Content-Type': content_type
  },
  'url': url,
  'method': method,
  'content': 'Evil unhashed content'
}

# Whoops! The developer forgot to pass in the `content` keyword. This could have been a mistake.
receiver = Receiver(
  lookup_credentials,
  request['headers']['Authorization'],
  request['url'],
  request['method'],
  content_type=request['headers']['Content-Type'])
```